### PR TITLE
fix #2142: migrated guarantor related fragments to viewbinding

### PR DESCRIPTION
**Pull request by PratyushSingh07**
Fixes #2142 

https://github.com/openMF/mifos-mobile/assets/90026952/4c9ce13f-19c2-401c-802f-92381f28e6b3

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
do review this @gururani-abhishek !!

File: app/src/main/java/org/mifos/mobile/ui/fragments/AddGuarantorFragment.kt [hunk 1]
```diff
@@ -171,10 +164,10 @@ class AddGuarantorFragment : BaseFragment(), AddGuarantorView {
...
-        tilFirstName?.editText?.setText(payload?.firstname)
-        tilLastName?.editText?.setText(payload?.lastname)
-        tilOfficeName?.editText?.setText(payload?.officeName)
-        (guarantorTypeField?.editText as?  MaterialAutoCompleteTextView)?.setText(payload?.guarantorType?.value!!)
+        binding.tilFirstName.editText?.setText(payload?.firstname)
+        binding.tilLastName.editText?.setText(payload?.lastname)
+        binding.tilOfficeName.editText?.setText(payload?.officeName)
+        (binding.guarantorTypeField.editText as?  MaterialAutoCompleteTextView)?.setText(payload?.guarantorType?.value!!)
 
     }
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/AddGuarantorFragment.kt]: It seems findPreviouslySelectedGurantorType() is not being used anywhere, can be removed after a thorough usage check.

File: app/src/main/java/org/mifos/mobile/ui/fragments/GuarantorDetailFragment.kt [hunk 2]
```diff
@@ -113,19 +93,19 @@ import javax.inject.Inject
...
-                            })
-                    .setNegativeButton(R.string.cancel)
-                    .createMaterialDialog()
-                    .show()
+                .init(context)
+                .setTitle(R.string.delete_guarantor)
+                .setMessage(getString(R.string.dialog_are_you_sure_that_you_want_to_string,
+                    getString(R.string.delete_guarantor)))
+                .setPositiveButton(getString(R.string.yes),
+                    DialogInterface.OnClickListener { _, _ ->
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/GuarantorDetailFragment.kt]: This redundant SAM constructor can be reformat. Please have a look.

**Review by gururani-abhishek — COMMENTED** [review]
Looks solid!! good job.

File: app/src/main/java/org/mifos/mobile/ui/fragments/GuarantorDetailFragment.kt (hunk 2 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/GuarantorDetailFragment.kt]: done

File: app/src/main/java/org/mifos/mobile/ui/fragments/AddGuarantorFragment.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/AddGuarantorFragment.kt]: done @gururani-abhishek  , I have also used `with(binding)`

**Comment by PratyushSingh07** [comment]
Hey @jawidMuhammadi kindly review this as well

**Review by jawidMuhammadi — APPROVED** [review]
LGTM!

**Pull request merged**